packages/tpetra/core/src/Tpetra_Details_FixedHashTable_def.hpp

```diff
@@ -1196,6 +1205,15 @@ init (const keys_type& keys,
     }
   }
 
+  // FENCE REVIEW - NOW PASSING
+  //   Testing: This code is exercised by unit tests.
+  //   GTX960:  Passed with CUDA_LAUNCH_BLOCKING=0 and fence removed.
+  //   White:   Passed with CUDA_LAUNCH_BLOCKING=0 and fence removed.
+  //   Plan:    Remove this fence if we refactor code below to avoid UVM access.
+  //   Notes:   Need to check debug path as well.
+  //            Seems the passes are false oks - this code could fail.
+  //            For example see Mark's comments: 'This assumes UVM'
+
```

Comment:
@kddevin Before this fence, it calls computeOffsetsFromCounts, which uses parallel scan. If scan does not apply fence, then we need the fence. 

@crtrott Do we apply fence at the end of parallel scan ? This scan does not request the reduction (the last scanned value)

Reply:
On Kokkos' slack, @crtrott answered "no" regarding Kokkos' applying a fence at the end of this parallel scan.

Reply:
Then this fence is necessary as we need to make sure that the offset view completes before the view is used in the next functor.

Reply:
Updated comments #7546